clients/include/testing_csrrf_refactlu.hpp

```diff
@@ -0,0 +1,604 @@
+                                                           &nnzURes,
+                                                           &hptrURes,
+                                                           &hindURes,
+                                                           &hvalURes));
+
+    EXPECT_EQ(nnzLRes, nnzL) << "where b = " << 0;
+    EXPECT_EQ(nnzURes, nnzU) << "where b = " << 0;
+    if(nnzLRes == nnzL && nnzURes == nnzU)
+    {
+        double errorL, errorU;
+        errorL   = norm_error('F', 1, nnzLRes, 1, hvalL[0], hvalLRes);
+        errorU   = norm_error('F', 1, nnzURes, 1, hvalU[0], hvalURes);
+        *max_err = max({*max_err, errorL, errorU});
+    }
+    else
+        *max_err = 1;
+#endif
+}
+
+template <typename T, typename Td, typename Ud, typename Th, typename Uh>
+void csrrf_refactlu_getPerfData(hipsolverRfHandle_t handle,
+                                const int           n,
+                                const int           nnzA,
+                                Ud&                 dptrA,
+                                Ud&                 dindA,
+                                Td&                 dvalA,
+                                const int           nnzL,
+                                Ud&                 dptrL,
+                                Ud&                 dindL,
+                                Td&                 dvalL,
+                                const int           nnzU,
+                                Ud&                 dptrU,
+                                Ud&                 dindU,
+                                Td&                 dvalU,
+                                Ud&                 dpivP,
+                                Ud&                 dpivQ,
+                                Uh&                 hptrA,
+                                Uh&                 hindA,
+                                Th&                 hvalA,
+                                Uh&                 hptrL,
+                                Uh&                 hindL,
+                                Th&                 hvalL,
+                                Uh&                 hptrU,
+                                Uh&                 hindU,
+                                Th&                 hvalU,
+                                Th&                 hvalT,
+                                Uh&                 hpivP,
+                                Uh&                 hpivQ,
+                                double*             gpu_time_used,
+                                double*             cpu_time_used,
+                                const int           hot_calls,
+                                const bool          perf,
+                                const fs::path      testcase)
+{
+    *cpu_time_used = nan(""); // no timing on cpu-lapack execution
+    *gpu_time_used = nan(""); // no timing on gpu-lapack execution
```

Comment:
we could be doing the setup, analysis and resetvalues only once, and timing the actual call to refRefactor. Not a big deal, something to think about.